Add `Releases` tab to the repo page

hkdobrev opened the pull request:
Closes #8

I've successfully added a releases tab.
However I have some concerns which I hope I'd be able to address with your help.

- Adding just `<span class="octicon octicon-tag">` does not render an icon. I needed to add a full SVG markup like the rest of the icons I could found on github.com.
- Appending the *Releases* tab and activating it instead of *Code* is rather slow and you could visually see the change after the page loads. Feedback appreciated.
- If you have an idea how to easily count the number of releases (or tags) to put next to the label on the tab, that would be great.

P.S. This also adds a new shortcut `g r` for navigating to the releases page of a repo :smiley_cat:

sindresorhus commented on extension/content.js:
```diff
@@ -59,6 +83,7 @@ document.addEventListener('DOMContentLoaded', () => {
 
 	if (isRepo) {
 		gitHubInjection(window, () => {
```
```js
gitHubInjection(window, {wait: 1}, () => {
```

To make it appear faster.

sindresorhus commented on extension/content.js:
```diff
@@ -26,6 +28,28 @@ function linkifyBranchRefs() {
+			'&nbsp;<span itemprop="name">Releases</span>' +
+			'<meta itemprop="position" content="6">' +
+		'</a>' +
+	'</span>';
```
Just use a template literal for the whole thing.

sindresorhus commented on extension/content.js:
```diff
@@ -26,6 +28,28 @@ function linkifyBranchRefs() {
 }
 
+function addReleasesTab() {
+	const releasesTabTemplate = '<span itemscope="" itemtype="http://schema.org/ListItem" itemprop="itemListElement">' +
```
We don't need this `itemscope="" itemtype="http://schema.org/ListItem" itemprop="itemListElement"`

sindresorhus commented on extension/content.js:
```diff
@@ -3,8 +3,10 @@
 const path = location.pathname;
 const isDashboard = path === '/';
 const isRepo = /^\/[^/]+\/[^/]+/.test(path);
+const vendorName = path.split('/')[1];
```
`vendorName` => `username`?

sindresorhus commented:
Nice work!

When I click the tab and then click the back button in the browser, the Releases tab is duplicated. We need to somehow guard it. @stefanbuck Any tips on how to handle this? Something the `injection` module could handle for us maybe?

hkdobrev commented:
@sindresorhus I think I've addressed your comments. I've also added a quick check if the Releases tab is not already present which solves the back button behaviour. It should also work if GitHub decides to add a Releases tab themselves :smile:

hkdobrev commented:
I've also fixed the activation of Releases tab when visiting the Tags page, e.g. https://github.com/sindresorhus/refined-github/tags.

stefanbuck commented:
GitHub.com use some kind of cache for pages that you visited before. So if you navigate back and forth you just switch between the cache pages. That's the reason why the button was added multiple times, because in the cache version it was already there. 

The quick check that @hkdobrev implemented looks simple and solid :+1:

sindresorhus commented:
@stefanbuck Thanks for chiming in. I'm aware of the caching. I was wondering more in the direction of a general solution than having to check everything you add.

stefanbuck commented:
Ahh I see, let me think about it.

sindresorhus commented:
@stefanbuck Ok, cool. Let's continue in https://github.com/octo-linker/injection/issues/2.

sindresorhus commented:
Perfect! Thank you @hkdobrev. This is a really nice addition.

Adding you to the repo per the [open open source idea](https://github.com/Level/community/blob/master/CONTRIBUTING.md) because of this awesome contribution.

> If you have an idea how to easily count the number of releases (or tags) to put next to the label on the tab, that would be great.

We could scrape it from the repo landing page when the user goes there, persist it in localstorage, and then have it available to any repo page. Not sure whether it's worth the trouble, though.
